install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml

```diff
@@ -11,9 +11,6 @@ rules:
   # managed by Cilium
   - pods
   verbs:
-  - get
-  - list
-  - watch
   - delete
```

Comment:
```patch
--- a/install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml
+++ b/install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml
@@ -7,14 +7,18 @@ rules:
 - apiGroups:
   - ""
   resources:
-  # to automatically delete [core|kube]dns pods so that are starting to being
-  # managed by Cilium
   - pods
   verbs:
   - get
   - list
   - watch
+{{- if hasKey .Values "disableEndpointCRD"}}
+{{- if (eq (.Values.disableEndpointCRD | quote ) ( "false" | quote )) }}
+  # to automatically delete [core|kube]dns pods so that are starting to being
+  # managed by Cilium
   - delete
+{{- end }}
+{{- end }}
 {{- if or .Values.operator.removeNodeTaints .Values.operator.setNodeNetworkStatus }}
 - apiGroups:
   - ""
diff --git a/operator/main.go b/operator/main.go
index 64db8943e4..fec16f8711 100644
--- a/operator/main.go
+++ b/operator/main.go
@@ -369,6 +369,8 @@ func onOperatorStartLeading(ctx context.Context) {
        // Restart kube-dns as soon as possible since it helps etcd-operator to be
        // properly setup. If kube-dns is not managed by Cilium it can prevent
        // etcd from reaching out kube-dns in EKS.
+       // If this logic is modified, make sure the operator's clusterrole logic for
+       // pods/delete is also up-to-date.
        if option.Config.DisableCiliumEndpointCRD {
                log.Infof("KubeDNS unmanaged pods controller disabled as %q option is set to 'disabled' in Cilium ConfigMap", option.DisableCiliumEndpointCRDName)
        } else if operatorOption.Config.UnmanagedPodWatcherInterval != 0 {
```

Reply:
I guess that's better. Feels a bit weird to put this onto `DisableCiliumEndpointCRD` though.

Could we instead just expose a "autoRestartDNS" option, and if it's false then we set `UnmanagedPodWatcherInterval=0` and drop these privileges?

Reply:
or maybe it could be more structured like this, I dunno:

```
unmanagedPodWatcher:
  restartDNS: true
  interval: Ns
```

Reply:
> Feels a bit weird to put this onto `DisableCiliumEndpointCRD` though.

To paint a bigger picture, I know that we have a scalability guide that recommends this option. Beyond that, it's a complicated option to explain to someone, I don't know if we describe the implications of disabling this CRD in the docs. At the very least it presumably makes debuggability more difficult because you can't list these objects any more. Disabling a chunk of functionality like this is also not really a user-facing "feature", it's an advanced tweak. If it's possible to disable this, why not disable it always? And in that case why are we making Cilium so complicated for users by forcing them to understand the deep architectural details of Cilium well enough to consider this option?

Reply:
Done, PTAL